lib/active_merchant/billing/gateways/visanet_peru.rb

```diff
@@ -116,8 +114,9 @@ def add_antifraud_data(params, options)
           antifraud[:billTo_postalCode] = billing_address[:zip]
         end
 
-        antifraud[:deviceFingerprintId] = options[:device_fingerprint_id]
-        antifraud[:merchantDefineData] = options[:merchant_define_data]
+        antifraud[:deviceFingerprintId] = options[:device_fingerprint_id] || SecureRandom.hex(16)
```

Comment:
@rwdaigle 

yes, although I didn't try NOT passing it.

![image](https://cloud.githubusercontent.com/assets/4900443/14689684/08aec308-0766-11e6-936e-31cf1f80f96c.png)